File: problems/2215/2215.find-the-difference-of-two-arrays.go
/*
 * @lc app=leetcode id=2215 lang=golang
 *
 * [2215] Find the Difference of Two Arrays
 */

package solution

// @lc code=start
func findDifference(nums1 []int, nums2 []int) [][]int {
	s1 := make(map[int]bool)
	s2 := make(map[int]bool)
	answer := make([][]int, 2)

	for _, v := range nums1 {
		s1[v] = true
	}
	for _, v := range nums2 {
		s2[v] = true
	}

	for v := range s1 {
		if _, ok := s2[v]; !ok {
			answer[0] = append(answer[0], v)
		}
	}
	for v := range s2 {
		if _, ok := s1[v]; !ok {
			answer[1] = append(answer[1], v)
		}
	}

	return answer
}

// @lc code=end

File: problems/2215/2215.find-the-difference-of-two-arrays.py
#
# @lc app=leetcode id=2215 lang=python3
#
# [2215] Find the Difference of Two Arrays
#


# @lc code=start
class Solution:
    def findDifference(
        self, nums1: List[int], nums2: List[int]
    ) -> List[List[int]]:
        # answer[0] -> all distinct integers in nums1 not in nums2
        # answer[1] -> all distinct integers in nums2 not in nums1

        ans1, ans2 = [], []
        set1, set2 = set(nums1), set(nums2)
        for val in set1:
            if val not in set2:
                ans1.append(val)
        for val in set2:
            if val not in set1:
                ans2.append(val)

        return [ans1, ans2]


# @lc code=end
